docs/reference/how-to/indexing-speed.asciidoc

```diff
@@ -0,0 +1,93 @@
+[[tune-for-indexing-speed]]
+== Tune for indexing speed
+
+=== Use bulk requests
+
+Bulk requests will yield much better performance than single-document index
+requests. In order to know the optimal size of a bulk request, you shoud run
+a benchmark on a single node with a single shard. First try to index 100
+documents at once, then 200, then 400, etc. doubling the number of documents
+in a bulk request in every benchmark run. When the indexing speed starts to
+plateau then you know you reached the optimal size of a bulk request for your
+data. In case of tie, it is better to err in the direction of too few rather
+than too many documents.
+
+=== Use multiple workers/threads to send data to elasticsearch
+
+A single thread sending bulk requests is unlikely to be able to max out the
+indexing capacity of an elasticsearch cluster. In order to use all resources
+of the cluster, you should send data from multiple threads or processes.
+
+Make sure to watch for `TOO_MANY_REQUESTS (429)` response codes
+(`EsRejectedExecutionException` with the Java client), which is the way that
+elasticsearch tells you that it cannot keep up with the current indexing rate.
+When it happens, you should pause ndexing a bit before trying again, ideally
+with exponential backoff.
+
+=== Disable swapping
+
+You should make sure that the operating system is not swapping out the java
+process by <<setup-configuration-memory,disabling swappping>>.
+
+=== Give memory to the filesystem cache
+
+The filesystem cache will be used in order to buffer I/O operations. You should
+make sure to give at least half the memory of the machine running elasticsearch
+to the filesystem cache.
+
+=== Use faster hardware
+
+If indexing is I/O bound, you should investigate giving more memory to the
+filesystem cache (see above) or buying faster drives. In particular SSD drives
+are known to perform better than spinning disks. Always use local storage,
+remote filesystems such as `NFS` or `SMB` should be avoided. Also beware of
+virtualized storage such as Amazon's `Elastic Block Storage`. Virtualized
+storage works very well with Elasticsearch, and it is appealing since it is so
+fast and simple to set up, but it is also unfortunately inherently slower on an
+ongoing basis when compared to dedicated local storage. In a recent informal
```

Comment:
The test came from a user report at the time who seemed quite informed, but I think we should remove reference to it here since I never cleanly reproduced the results (though I still expect EBS to be slower than local SSD).

Instead of referencing the "informal test", change this to state that if you do put an index on EBS to be sure to used provisioned IOPs otherwise your operations can be quickly throttled.